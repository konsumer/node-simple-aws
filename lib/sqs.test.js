/* global describe, it, expect, jest */

const { SQS } = require('..')

const sqs = new SQS()

describe('sqs', () => {
  it('should have unit-tests', () => {
    expect(1 + 1).toBe(2)
  })

  it('should be able to send a message', async () => {
    sqs.client.sendMessage = jest.fn((params, cb) => {
      expect(params).toMatchSnapshot()
      cb(null, true)
    })
    await new Promise((resolve) => {
      sqs.sendMessage('TEST', 'testing', (err, res) => {
        expect(sqs.client.sendMessage.mock.calls.length).toBe(1)
        expect(err).toBe(null)
        expect(res).toBe(true)
        sqs.client.sendMessage.mockReset()
        resolve()
      })
    })
  })
})
