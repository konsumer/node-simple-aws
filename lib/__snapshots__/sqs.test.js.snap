// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqs should be able to send a message 1`] = `
Object {
  "MessageBody": "testing",
  "QueueUrl": "TEST",
}
`;
